features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/MarkerCacheIT.java

```diff
@@ -221,11 +232,97 @@ public void testNMS12740() throws Exception {
 
             elasticFlowRepository.persist(Lists.newArrayList(getMockFlow(Flow.Direction.EGRESS)), getMockFlowSource());
 
-            Assert.assertEquals(0,snmpInterfaceDao.findAllHavingIngressFlows(2).size());
-            Assert.assertEquals(0, snmpInterfaceDao.findAllHavingEgressFlows(2).size());
+            assertEquals(0, snmpInterfaceDao.findAllHavingIngressFlows(2).size());
+            assertEquals(0, snmpInterfaceDao.findAllHavingEgressFlows(2).size());
 
             // the following call resulted to two wrong entries before, since the wrong query returned entries from other nodes with egress flows
-            Assert.assertEquals(0, snmpInterfaceDao.findAllHavingFlows(2).size());
+            assertEquals(0, snmpInterfaceDao.findAllHavingFlows(2).size());
         }
     }
+
+    @Test
+    public void shouldDistinguishBetweenIngressAndEgressWhenDeterminingIfFlowsAreAvailable() throws Exception {
+        Assert.assertFalse(OnmsSnmpInterface.INGRESS_AND_EGRESS_REQUIRED);
+
+        stubFor(post("/_bulk")
+                .willReturn(aResponse()
+                        .withStatus(200)
+                        .withHeader("Content-Type", "application/json")));
+
+        final ClassificationEngine classificationEngine = new DefaultClassificationEngine(() -> Lists.newArrayList(
+                new RuleBuilder().withName("http").withDstPort("80").withProtocol("tcp,udp").build(),
+                new RuleBuilder().withName("https").withDstPort("443").withProtocol("tcp,udp").build()
+        ), FilterService.NOOP);
+
+        final DocumentEnricher documentEnricher = new DocumentEnricher(
+                new MetricRegistry(), nodeDao, interfaceToNodeCache, sessionUtils, classificationEngine,
+                new CacheConfigBuilder()
+                        .withName("flows.node")
+                        .withMaximumSize(1000)
+                        .withExpireAfterWrite(300)
+                        .build());
+
+        final JestClientFactory factory = new JestClientFactory();
+        factory.setHttpClientConfig(new HttpClientConfig.Builder("http://localhost:" + wireMockRule.port()).build());
+
+        try (JestClient client = factory.getObject()) {
+            final ElasticFlowRepository elasticFlowRepository = new ElasticFlowRepository(new MetricRegistry(),
+                    client, IndexStrategy.MONTHLY, documentEnricher,
+                    sessionUtils, nodeDao, snmpInterfaceDao,
+                    new MockIdentity(), new MockTracerRegistry(), new MockDocumentForwarder(), new IndexSettings(),
+                    mock(SmartQueryService.class));
+
+            Integer ingress = 2;
```

Comment:
Any reason why these (and others below) are `Integer`, not `int`?

Reply:
In this case: just inline them or make them real constants (same for the `nodeId` below.

Reply:
> Any reason why these (and others below) are `Integer`, not `int`?
nope. fixed.

Reply:
> In this case: just inline them or make them real constants (same for the `nodeId` below.

The reason I wanted to use a named variable (and not inline) is that it reads easier (2 has no meaning). 
But they are only used in the scope of the method so I didn't wanted to make it a class constant.
I made it final as a compromise. Sounds good?